src/main/java/org/lmdbjava/Dbi.java

```diff
@@ -249,9 +249,10 @@ public String getName() {
    * {@link Cursor#renew(org.lmdbjava.Txn)} before finally closing it.
    *
    * @param txn transaction handle (not null; not committed)
+   * @param <B> The buffer type
    * @return cursor handle
    */
-  public Cursor<T> openCursor(final Txn<T> txn) {
+  public <B> Cursor<B> openCursor(final Txn<B> txn) {
```

Comment:
The buffer type of the cursor is not dependent on the buffer type of the Dbi, only the Txn.